engine/src/main/java/org/terasology/rendering/nui/editor/utils/NUIEditorMenuTreeBuilder.java

```diff
@@ -250,7 +252,7 @@ private void populateContextMenu(JsonTree node, MenuTree addTree, boolean isSkin
                 // If the node is part of a layout, add an option to add the layoutInfo node.
                 if (nodeInfo.getLayoutClass() != null) {
                     String layoutInfo = "layoutInfo";
-                    if (!node.hasChildWithKey(layoutInfo)) {
+                    if (!Objects.equals(node.getValue().getKey(), layoutInfo) && !node.hasChildWithKey(layoutInfo)) {
```

Comment:
There are other cases too where we don't want to offer a layoutInfo as option that can be added.

e.g. the context menu for a position-horizontal-center still offers with that change to add a "layoutInfo" which is similarly wrong.
Some classes to look at: 
* UIWidgetTypeAdapter.deserialize for the actual way of how layoutInfo gets used
* NUIEditorNodeUtils.getNodeInfo for logic you wrote that does something in the direction of detecting what is what.